doorkeeper/worker/DoorkeeperRevisionFeedWorker.php

```diff
@@ -0,0 +1,227 @@
+          break;
+
+        // https://trello.com/c/s7sauVAO/286-3-trigger-the-plan-changes-or-request-review-actions-clears-all-r-flags-on-the-revision-attachment-in-the-associated-bug
+        case self::REVISION_PLAN_CHANGES:
+        case self::REVISION_REVIEW_REQUEST:
+          $this->clear_all_review_statuses($revision);
+          break;
+
+        // https://trello.com/c/0bbNOGlC/386-3-obsolete-stub-attachment-from-bmo-if-associated-revision-is-abandoned
+        case self::REVISION_ABANDONED:
+          $this->obsolete_attachment($revision);
+          break;
+
+        default:
+          echo pht('David, you forgot to cover %s', $transaction_type);
+          break;
+      }
+
+      echo '------ /DoorkeeperRevisionFeedWorker ------';
+    }
+
+    private function update_review_statuses($revision) {
+      $accepted_bmo_ids = array();
+
+      // Grab all reviewers with an "accepted" status
+      $reviewers = $revision->getReviewers();
+      $accepted_phids = array();
+      foreach($reviewers as $reviewer) {
+        // NOTE:  There's an "STATUS_ACCEPTED_OLDER"...what does that represent?
+        // https://github.com/phacility/phabricator/blob/48a74de0b64901538feb878e2f12e18e605ca76a/src/applications/differential/editor/DifferentialTransactionEditor.php#L215
+        if($reviewer->getReviewerStatus() === DifferentialReviewerStatus::STATUS_ACCEPTED) {
+          $accepted_phids[] = $reviewer->getReviewerPHID();
+        }
+      }
+
+      // Use the External User Query to get their BMO IDS
+      if(count($accepted_phids)) {
+        $bmo_users = id(new PhabricatorExternalAccountQuery())
+          ->setViewer(PhabricatorUser::getOmnipotentUser())
+          ->withAccountTypes(array(PhabricatorBMOAuthProvider::ADAPTER_TYPE))
+          ->withUserPHIDs($accepted_phids)
+          ->execute();
+        foreach($bmo_users as $user) {
+          $accepted_bmo_ids[] = $user->getAccountID();
+        }
+      }
+
+      $this->send_update_request($revision, $accepted_bmo_ids);
+    }
+
+
+    private function clear_all_review_statuses($revision) {
+      $this->send_update_request($revision, array());
+    }
+
+    private function obsolete_attachment($revision) {
+      $request_data = array(
+        'revision_id' => $revision->getID(),
+        'bug_id' => $this->get_bugzilla_bug_id($revision)
+      );
+
+      $future_uri = id(new PhutilURI(PhabricatorEnv::getEnvConfig('bugzilla.url')))
+        ->setPath('/rest/phabbugz/obsolete/');
+
+      $future = $this->get_http_future($future_uri)
+        ->setMethod('PUT')
+        ->setData($request_data)
+        ->setExpectStatus(array(200));
+
+      // Debugging
+      echo pht('Making a request to: %s', (string) $future_uri);
+      echo 'Using data:'.json_encode($request_data, JSON_FORCE_OBJECT);
+
+      try {
+        list($status) = $future->resolve();
+        if($status->getStatusCode() != 200) {
+          // TODO:  What should we do in case of failure?  Re-queue?
```

Comment:
Does Doorkeeper have a way to re-queue built-in that doesn't require republishing to the feed?